sql/sql_parse.cc

```diff
@@ -3203,6 +3206,60 @@ int mysql_execute_command(THD *thd, bool first_level) {
       res = purge_master_logs_before_date(thd, purge_time);
       break;
     }
+    case SQLCOM_PURGE_UUID: {
+      if (check_global_access(thd, SUPER_ACL)) goto error;
+      rpl_sidno server_sidno = gtid_state->get_server_sidno();
+      rpl_sidno master_sidno = 0;
+      // We should not purge gtids received from current active master.
+      get_master_sidno(thd, master_sidno);
+      rpl_sidno_container sidnos;
+      char *token = strtok(lex->gtid_string, ",");
+      while (token) {
+        rpl_sid sid;
+        while (my_isspace(&my_charset_utf8_general_ci, *token)) token++;
+        auto len = strlen(token);
```

Comment:
Malformed gtid_strings with double commas like "uuid1:1-3,,uuid2:3-5" could result in len = 0. I'll can fix this before applying the change.

Reply:
Fixed, and added a test for this.